routes/user/presentations/presentation/handlers.js

```diff
@@ -318,180 +316,52 @@ var getPresentationStats = gen.lift(function *getPresentationStats(req, res, nex
-        exercise.names = ['maxNumSubmissions', 'confidence'];
-        exercise.maxNumSubmissions = exObject.maxNumSubmissions;
-        exercise.confidence = exObject.confidence;
-        slide.exercises.push(exercise);
-      }
-      data.push(slide)
-    }
-  }
-  return data.reverse();
-});
-
-var isSlideshowActiveByUser = coroutine(function* isSlideshowActiveByUser(slideshowId, userId) {
-  var query = {
-    'presenter': userId,
-    'slides': slideshowId,
-    'endDate': null
-  }
-  var sesstions = yield Session.find(query).exec();
-  if ( !sesstions || sesstions.length == 0 ) {
-    return { flag: false }
-  }
-  return { flag: true, sessions: sesstions }
-});
-
-
-var putPresentationSettings = coroutine(function* putPresentationSettingsGen(req, res) {
-  return res.json({msg: "Alles gut"});
-});
-
-var configurePresentation = coroutine(function* configurePresentation(req, res) {
+  var user        = req.user;
+  var userId      = user._id;
+  var username    = user.username;
   var slideshowId = req.params.presentationId;
-  var slideshow;
+
   try{
-    var slideshow = yield Slideshow.findById(slideshowId).exec();
-  }catch(err){
+    slideshow = yield Slideshow.findById(slideshowId).exec();
+  } catch(err){
     logger.error("Presentation %s not found", req.params.presentationId);
     logger.error(err.message, { err: err.stack });
     res.status(404);
     return res.render('404', {'msg': 'Presentation not found'});
   }
 
-  var slideConf = yield getConf(slideshow.configuration);
-  var data = yield transform(slideshow.exercisesPerSlide);
-  var username = req.user.username;
-
-  var param = {
-    title: slideshow.title,
-    username: username,
-    slideshowId: slideshowId,
-    slideConf: slideConf,
-    data: data
-  }
+  var presentationSettings = yield slideshow.getSettings();
+  var exerciseSettings = yield settings.getDustifySettingsOfExercisesAll(slideshowId);
 
   // Whether the slideshow is currently active(running) by this user
-  var result = yield isSlideshowActiveByUser(slideshowId, req.user._id);
-  if ( !result.flag ) {
-    res.render('presentationSettings', param);
-  } else {
-    var sessionId = result.sessions[0]._id;
-    var livelink = ['/', req.user.username,'/presentations/', slideshowId, '/live/', sessionId, '/?role=presenter&view=presentation'].join('');
-    param.livelink = livelink;
-    res.render('presentationSettingsRuntime', param);
-  }
-   
-});
-
-var configurePresentationSaveExercise = coroutine(function* configurePresentationSaveExercise(req, res) {
-  var exerciseId = req.body.uid;
-  var slideshowId = req.params.presentationId;
-  // TODO: not to hardcode
-  // 
-  var conf = {
-    maxNumSubmissions: Number(req.body.max) ? Number(req.body.max) : 0,
-    confidence: req.body.hasOwnProperty('confidence')
-  }
-
-  var state = yield Conf.updateExerciseConf(slideshowId, exerciseId, conf);
+  var sessionId = yield presUtils.isLiveBy(userId, slideshowId);
```

Comment:
looks like this is work for the `isRouteOwner` middleware: https://github.com/ASQ-USI/ASQ/blob/master/lib/middleware/middleware.js#L161